device.go

```diff
@@ -45,6 +45,10 @@ type DeviceDesc struct {
 
 	// Configuration information
 	Configs map[int]ConfigDesc
+
+	IdxSerial       int   // The index into the getStringDescriptor() to get the Serial Number
```

Comment:
I'd like to keep the API of the library disconnected from USB implementation details, instead focusing on ease of use. The user should not need to know how the string descriptors work and perhaps we can add a better interface here.

One idea would be to extract these descriptors into the DeviceDesc immediately during enumeration, so that we have a "SerialNumber, ProductName, Manufacturer string" fields. We can similarly add configuration/interface "Description" field from iConfiguration/iInterface.

An alternative would be to make descriptor indices unexported fields and add methods that will wrap GetStringDescriptor calls, i.e. func (*Device) SerialNumber() string, but that would require an open device handle. It seems populating at enumeration time is better.

WDYT?